File: dequad/intde2t.f
!     test of intde2.f
!
      include 'intde2.f'
!
      program main
      implicit real*8 (a - i, o - z)
      parameter (lenaw = 8000)
      parameter (tiny = 1.0d-307)
      dimension aw(0 : lenaw - 1)
      external f1, f2, f3, f4, f5, f6
      common  / nfunc / nn
!     
      call intdeini(lenaw, tiny, 1.0d-15, aw)
      nn = 0
      call intde(f1, 0.0d0, 1.0d0, aw, i, err)
      write (*, *) 'I_1=int_0^1 1/sqrt(x) dx'
      write (*, *) '  I_1=', i, ',  err=', err, ',  n=', nn
      nn = 0
      call intde(f2, 0.0d0, 2.0d0, aw, i, err)
      write (*, *) 'I_2=int_0^2 sqrt(4-x*x) dx'
      write (*, *) '  I_2=', i, ',  err=', err, ',  n=', nn
!     
      call intdeiini(lenaw, tiny, 1.0d-15, aw)
      nn = 0
      call intdei(f3, 0.0d0, aw, i, err)
      write (*, *) 'I_3=int_0^infty 1/(1+x*x) dx'
      write (*, *) '  I_3=', i, ',  err=', err, ',  n=', nn
      nn = 0
      call intdei(f4, 0.0d0, aw, i, err)
      write (*, *) 'I_4=int_0^infty exp(-x)/sqrt(x) dx'
      write (*, *) '  I_4=', i, ',  err=', err, ',  n=', nn
!     
      call intdeoini(lenaw, tiny, 1.0d-15, aw)
      nn = 0
      call intdeo(f5, 0.0d0, 1.0d0, aw, i, err)
      write (*, *) 'I_5=int_0^infty sin(x)/x dx'
      write (*, *) '  I_5=', i, ',  err=', err, ',  n=', nn
      nn = 0
      call intdeo(f6, 0.0d0, 1.0d0, aw, i, err)
      write (*, *) 'I_6=int_0^infty cos(x)/sqrt(x) dx'
      write (*, *) '  I_6=', i, ',  err=', err, ',  n=', nn
      end
!
!
      function f1(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f1 = 1 / sqrt(x)
      end
!
!
      function f2(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f2 = sqrt(4 - x * x)
      end
!
!
      function f3(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f3 = 1 / (1 + x * x)
      end
!
!
      function f4(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f4 = exp(-x) / sqrt(x)
      end
!
!
      function f5(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f5 = sin(x) / x
      end
!
!
      function f6(x)
      implicit real*8 (a - i, o - z)
      common  / nfunc / n
      n = n + 1
      f6 = cos(x) / sqrt(x)
      end
!
!

File: dequad/intde2t.c
/* test of intde2.c */

#include <math.h>
#include <stdio.h>

#include "intde2.c"

int nfunc;

main()
{
    double f1(double), f2(double), f3(double), f4(double), 
        f5(double), f6(double);
    void intdeini(int lenaw, double tiny, double eps, double *aw);
    void intde(double (*f)(double), double a, double b, 
        double *aw, double *i, double *err);
    void intdeiini(int lenaw, double tiny, double eps, double *aw);
    void intdei(double (*f)(double), double a, double *aw, 
        double *i, double *err);
    void intdeoini(int lenaw, double tiny, double eps, 
        double *aw);
    void intdeo(double (*f)(double), double a, double omega, 
        double *aw, double *i, double *err);
    int lenaw;
    extern int nfunc;
    double tiny, aw[8000], i, err;
    
    lenaw = 8000;
    tiny = 1.0e-307;
    
    intdeini(lenaw, tiny, 1.0e-15, aw);
    nfunc = 0;
    intde(f1, 0.0, 1.0, aw, &i, &err);
    printf("I_1=int_0^1 1/sqrt(x) dx\n");
    printf(" I_1= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
    nfunc = 0;
    intde(f2, 0.0, 2.0, aw, &i, &err);
    printf("I_2=int_0^2 sqrt(4-x*x) dx\n");
    printf(" I_2= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
    
    intdeiini(lenaw, tiny, 1.0e-15, aw);
    nfunc = 0;
    intdei(f3, 0.0, aw, &i, &err);
    printf("I_3=int_0^infty 1/(1+x*x) dx\n");
    printf(" I_3= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
    nfunc = 0;
    intdei(f4, 0.0, aw, &i, &err);
    printf("I_4=int_0^infty exp(-x)/sqrt(x) dx\n");
    printf(" I_4= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
    
    intdeoini(lenaw, tiny, 1.0e-15, aw);
    nfunc = 0;
    intdeo(f5, 0.0, 1.0, aw, &i, &err);
    printf("I_5=int_0^infty sin(x)/x dx\n");
    printf(" I_5= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
    nfunc = 0;
    intdeo(f6, 0.0, 1.0, aw, &i, &err);
    printf("I_6=int_0^infty cos(x)/sqrt(x) dx\n");
    printf(" I_6= %lg\t, err= %lg\t, N= %d\n", i, err, nfunc);
}


double f1(double x)
{
    extern int nfunc;
    
    nfunc++;
    return 1 / sqrt(x);
}


double f2(double x)
{
    extern int nfunc;
    
    nfunc++;
    return sqrt(4 - x * x);
}


double f3(double x)
{
    extern int nfunc;
    
    nfunc++;
    return 1 / (1 + x * x);
}


double f4(double x)
{
    extern int nfunc;
    
    nfunc++;
    return exp(-x) / sqrt(x);
}


double f5(double x)
{
    extern int nfunc;
    
    nfunc++;
    return sin(x) / x;
}


double f6(double x)
{
    extern int nfunc;
    
    nfunc++;
    return cos(x) / sqrt(x);
}

